test/e2e/upgrades/apps/daemonsets.go

```diff
@@ -68,6 +68,7 @@ func (t *DaemonSetUpgradeTest) Setup(f *framework.Framework) {
 						{
 							Name:            daemonSetName,
 							Image:           image,
+							Args:            []string{"serve-hostname"},
```

Comment:
same concern about assuming that `image` will be `agnhost` here

Reply:
The assumption is correct, the ``image`` variable is defined just above (https://github.com/kubernetes/kubernetes/pull/79238/commits/46b32be8bf06ba318e25e095f44b87c9571cc467#diff-8877b0fe1bb72e9deb2a6289ac74a846R46) and it points to ``frameworkServeHostnameImage``, which is now ``agnhost``.

I was going to rename ``frameworkServeHostnameImage`` to something else, but that would have generated a lot of useless churn which would have made it harder to review, and, the name ``frameworkServeHostnameImage`` still makes sense: it's an image which can serve hostname.